modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.jsx

```diff
@@ -179,44 +240,48 @@ const AppContextProvider = ({children}) => {
 			const user = await getUser(projectExternalReferenceCode);
 
 			if (user) {
-				const isValid = await isValidPage(
-					client,
+				const userProjectAccess = await getUserProjectAccess(
 					user,
-					projectExternalReferenceCode,
-					ROUTE_TYPES.project
+					projectExternalReferenceCode
 				);
-
-				if (isValid) {
-					let accountBrief = user.accountBriefs?.find(
-						(accountBrief) =>
-							accountBrief.externalReferenceCode ===
-							projectExternalReferenceCode
+	
+				if (userProjectAccess.hasProjectAccess) {
+					const isValid = await isValidPage(
+						client,
+						user,
+						projectExternalReferenceCode,
+						ROUTE_TYPES.project
 					);
 
-					const apiPermission = 
-						user.isOmniAdmin || user.isPartner || user.isStaff;
-
-					if (!accountBrief && apiPermission) {
-						const {data: dataAccount} = await client.query({
-							query: getAccountByExternalReferenceCode,
-							variables: {
-								externalReferenceCode: projectExternalReferenceCode,
+					if (isValid) {
+						let accountBrief = user.accountBriefs?.find(
+							(accountBrief) =>
+								accountBrief.externalReferenceCode ===
+								projectExternalReferenceCode
+						);
+
+						if (!accountBrief && !userProjectAccess.denyAccess) {
```

Comment:
As you added this verification (if the userProjectAccess exists in another file), I'm adding here too.

```suggestion
						if (!accountBrief && !userProjectAccess?.denyAccess) {
```

Reply:
In this section here, because the userProjectAccess object is being returned, it wouldn't ever be undefined there. Is ? still required?

Reply:
no, you can remove the ?

Reply:
keeping the same, thanks for confirming